src/components/views/context_menus/RoomGeneralContextMenu.tsx

```diff
@@ -45,13 +45,60 @@ import { useSettingValue } from "../../../hooks/useSettings";
 
 export interface RoomGeneralContextMenuProps extends IContextMenuProps {
     room: Room;
+    /**
+     * Called when the 'favourite' option is clicked, after the menu has processed
+     * whatever caused the click in the normal way.
+     * @param event The event that caused the click
```

Comment:
I'm sorry to keep banging on about this, but "whatever caused the click"? I mean, to me, a "click" is the physical movement of a mouse button, which is caused by someone's finger, I guess? Then there's a "click event", I suppose, which is a reflection of the fact a click has happened?

I think part of the problem is that this isn't just called when the option is *clicked*: you can also select it via the keyboard, for instance. So the name of the callback is misleading; we don't need to fix that now, but let's not propagate the confusion to the documentation.

Is this more accurate?

```suggestion
     * Called when the 'favourite' option is selected, after the menu has processed
     * the mouse or keyboard event.
     * @param event The event that caused the option to be selected.
```

Likewise with all the others, of course.

Reply:
Yeah, this probably is better. I feel like naming thing `onClick` is a fairly universal pattern where it's widely understood that it can be triggered via other means though?

Reply:
possibly. I think `on*Click` is ok as a name for a method. But I don't really think that's the same as redefining what the word "click" means.